clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -550,7 +550,7 @@ public boolean poll(Timer timer, boolean waitForJoinGroup) {
                     // refresh metadata before re-joining the group as long as the refresh backoff time has
                     // passed.
                     if (this.metadata.timeToAllowUpdate(timer.currentTimeMs()) == 0) {
-                        this.metadata.requestUpdate();
+                        this.metadata.requestUpdate(true);
```

Comment:
Why is permitBackoffOnEquivalentResponses set to true? We refresh the metadata here not because we have discovered stale metadata.

Reply:
OK. I thought this one was a bit debatable. I've changed to `false`.